app/models/request.rb

```diff
@@ -20,6 +20,12 @@ class Request < ApplicationRecord
 
   before_create :set_context
 
+  def switch_context
+    ManageIQ::API::Common::Request.with_request(context.transform_keys(&:to_sym)) do
+      ActsAsTenant.with_tenant(tenant) { yield }
```

Comment:
@bzwei This seems like its not in the right place, I had talked to @Fryguy to move this into the common gem so all paths be they come from the controller or background jobs should be able to set this

Reply:
This line is still needed. The common gem does not automatically switch tenant.